### PR TITLE
3581-2

### PR DESCRIPTION
**Pull request by elishved**
## Номер
3581

## Номер задания
2

## Ссылка на видео с демо
https://youtu.be/pnwFW5ZaXBE

### CODE REVIEW

**Comment by elishved** [comment]
@anatoliliotych please review

File: 3581/2/commands/command_start.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      respond_with :message, text: t(:already_registred)
+
+    else
+      save_context :number_input
+      respond_with :message, text: t(:not_registred)
+    end
+  end
+
+  def number_input(number)
+    if NumbersHandler.new(number, self).valid_number
```
Comment (andreydro) [inline 3581/2/commands/command_start.rb]: move this check to separate method

File: 3581/2/confirmation.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      save_context :geolocation
+      respond_with :message, text: t(:geo)
+    else
+      save_context :selfie_attachment
+      respond_with :message, text: t(:not_selfie)
+    end
+  end
+
+  def geolocation(*)
+    if geo?
```
Comment (andreydro) [inline 3581/2/confirmation.rb]: I don't like short names. Some people can not get what is `geo` from the first time

File: 3581/2/env.sample [hunk 3]
```diff
@@ -0,0 +1 @@
+TOKEN=844169864:AAHRq2KPWcpB10Nir
```
Comment (andreydro) [inline 3581/2/env.sample]: Add `env` file to `gitignore`, nobody should see it

File: 3581/2/numbers_handler.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    @number = number
+    @context = context
+  end
+
+  def valid_number
+    number_presence && number_is_not_in_use
+  end
+
+  def number_presence
+    File.open('./data/numbers.txt').each do |file_number|
```
Comment (andreydro) [inline 3581/2/numbers_handler.rb]: Assign `File.open` to variable and user `include?` to check number presence

File: 3581/2/start_bot.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+require './commands/command_checkout.rb'
+
+class WebhooksController < Telegram::Bot::UpdatesController
+  include Telegram::Bot::UpdatesController::MessageContext
+  include Telegram::Bot::UpdatesController::Session
+  include CommandStart
+  include CommandCheckin
+  include CommandCheckout
+
+  Telegram::Bot::UpdatesController.session_store = :redis_store, { expires_in: 2_592_000 }
```
Comment (andreydro) [inline 3581/2/start_bot.rb]: What does `2_592_000` mean? Move it to constant

File: 3581/2/start_bot.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  def process_action(*)
+    super
+  ensure
+    chat_session.commit
+  end
+end
+
+bot = Telegram::Bot::Client.new(ENV['TOKEN'])
+I18n.load_path << Dir[File.expand_path('config/local_msg') + '/*.yml']
+# poller-mode
```
Comment (andreydro) [inline 3581/2/start_bot.rb]: Do we need it in here?

File: 3581/2/env.sample (hunk 3 above)
Comment (elishved) [reply 3581/2/env.sample]: This is an incorrect token. This file is created for example. Correct token is stored in a file .env.

File: 3581/2/commands/command_start.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      respond_with :message, text: t(:already_registred)
+
+    else
+      save_context :number_input
+      respond_with :message, text: t(:not_registred)
+    end
+  end
+
+  def number_input(number)
+    if correct_number(number)
```
Comment (andreydro) [inline 3581/2/commands/command_start.rb]: Try to think about method names in way you speak. And if method return boolean, add `?` to the method name. 
`if number_is_correct?(number)` sounds better

**Pull request merged**